src/actions/vstudio/vs2010_nuget.lua

```diff
@@ -54,33 +37,148 @@
 		end
 	end
 
+	function nuget2010.packageAPIInfo(prj, package)
+		local id = nuget2010.packageId(package)
+		local version = nuget2010.packageVersion(package)
 
---
--- Generates the packages.config file.
---
+		if not packageAPIInfos[package] then
+			local packageAPIInfo = {}
 
-	function nuget2010.generatePackagesConfig(obj)
-		local wks = obj.workspace
+			printf("Examining NuGet package '%s'...", id)
+			io.flush()
 
-		local done = {}
-		local packages = {}
-		for prj in p.workspace.eachproject(wks) do
-			for i = 1, #prj.nuget do
-				local package = prj.nuget[i]
+			local response, err, code = http.get(string.format("https://api.nuget.org/v3/registration1/%s/index.json", id:lower()))
```

Comment:
I really like this to be configurable, there is other nuget streams out there, and certainly within companies like ours (Blizzard) we host our own internal caching streams, through things like artifactory, and myget...